### PR TITLE
Import Github Data

### PR DESCRIPTION
**Pull request by stevenelleman**
Review order: 
1. OAuth Token Issuance 
2. Strava Imports 
3. Github Imports

### CODE REVIEW

File: packages/types/src/oauth/github.ts [hunk 1]
```diff
@@ -38,4 +57,27 @@ export const GithubBearerTokenSchema = z.object({
...
+
+export type GithubUserResponse = z.infer<typeof GithubUserResponseSchema>;
+
+export function MapGithubCommitContributionsToLeaderboardEntry(username: string, chipIssuer: ChipIssuer, resp: any): LeaderboardEntry {
+
+  let commits: number = 0;
+
+  if (resp && resp.data && resp.data.user && resp.data.user.contributionsCollection && resp.data.user.contributionsCollection.totalCommitContributions) {
+    commits = resp.data.user.contributionsCollection.totalCommitContributions;
+  }
```
Comment (stevenelleman) [inline packages/types/src/oauth/github.ts:74]: While this is currently only returning the commit count my query has other contribution info: 
```
{
    "data": {
        "user": {
            "contributionsCollection": {
                "totalCommitContributions": 2,
                "totalIssueContributions": 1,
                "totalPullRequestContributions": 8,
                "totalPullRequestReviewContributions": 7
            }
        }
    }
}
```

I had an oddly low commit count from the last week, but with all contributions it looks about right (2 commits vs ~20 contributions). I wonder if the low commit number has to do with squashing commits?

File: apps/frontend/src/lib/oauth/imports.ts [hunk 2]
```diff
@@ -28,6 +34,63 @@ async function stravaFetch(token: AccessToken): Promise<Response> {
...
+  const rawUser = await user.json();
+
+  const resp: GithubUserResponse = GithubUserResponseSchema.parse(rawUser);
+
+  const username = resp.login;
+  // NOTE: with the exception of year, values are indices
+  const from = new Date(2024, 9, 21); // Oct 20, 2024
+  const to = new Date(2024, 9, 28); // Oct 27, 2024
+
+  // TODO: find graphgl library for handling query / variables
```
Comment (stevenelleman) [inline apps/frontend/src/lib/oauth/imports.ts:58]: I should make this safer / easier to deal with tomorrow by finding a graphql ts library.

File: apps/backend/.env.example [hunk 3]
```diff
@@ -3,4 +3,8 @@ DATABASE_URL="postgresql://postgres:postgres@localhost:5432/connections_test?sch
 AWS_ACCESS_KEY_ID="EXAMPLE_ACCESS_KEY_ID"
 AWS_SECRET_ACCESS_KEY="EXAMPLE_SECRET_ACCESS_KEY"
 AWS_REGION="us-east-1"
-AWS_SES_SENDER_EMAIL="test@gmail.com"
\ No newline at end of file
+AWS_SES_SENDER_EMAIL="test@gmail.com"
+
+# Github OAuth only in backend because that's where access token request originates
+NEXT_PUBLIC_OAUTH_GITHUB_CLIENT_ID=""
```
Comment (AndrewCLu) [inline apps/backend/.env.example:9]: remove NEXT_PUBLIC

File: apps/frontend/src/pages/oauth/exchange_token.tsx [hunk 4]
```diff
@@ -1,30 +1,132 @@
...
+
+async function getOAuthAccessToken(app: string, code: string, details: OAuthAppDetails): Promise<AccessToken | null> {
+  let token: AccessToken | undefined | null;
+
+  // First check if access token already exists and is active in local storage
+  token = await getAccessToken(app)
+  if (token) {
+    // NOTE: when there are more scopes, check token scope, XOR save token by $app_$scope
+
+    if (app == GITHUB) {
```
Comment (AndrewCLu) [inline apps/frontend/src/pages/oauth/exchange_token.tsx:33]: is this supposed to be ===

File: apps/frontend/src/config.ts [hunk 5]
```diff
@@ -22,19 +22,21 @@ export const OAUTH_APP_MAPPING: Record<string, OAuthAppDetails> = {
...
+        scope: "read",
       }
     ],
   },
   "github": {
     client_side_fetching: false,
-    can_import: false,
+    can_import: true,
     token_url: "https://github.com/login/oauth/access_token",
     redirect_uri: "https://connections.cursive.team/oauth/exchange_token&approval_prompt=force&scope=read",
```
Comment (AndrewCLu) [inline apps/frontend/src/config.ts:33]: replace with FRONTEND_URL

File: apps/frontend/src/lib/oauth/github.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    commits += resp.data.user.contributionsCollection.totalCommitContributions;
+  }
+  if (resp && resp.data && resp.data.user && resp.data.user.contributionsCollection && resp.data.user.contributionsCollection.totalIssueContributions) {
+    commits += resp.data.user.contributionsCollection.totalIssueContributions;
+  }
+  if (resp && resp.data && resp.data.user && resp.data.user.contributionsCollection && resp.data.user.contributionsCollection.totalPullRequestContributions) {
+    commits += resp.data.user.contributionsCollection.totalPullRequestContributions;
+  }
+  if (resp && resp.data && resp.data.user && resp.data.user.contributionsCollection && resp.data.user.contributionsCollection.totalPullRequestReviewContributions) {
+    commits += resp.data.user.contributionsCollection.totalPullRequestReviewContributions;
```
Comment (stevenelleman) [inline apps/frontend/src/lib/oauth/github.ts:22]: This is terrible, I'll fix this.

**Review by AndrewCLu — APPROVED** [review]

**Pull request merged**